### PR TITLE
Add features for media framework

### PR DESCRIPTION
**Pull request by jsdosa**
1. MediaPlayer - playback the media.
2. MediaRecorder - capture the media.
3. AudioManager - manage the audio device.
4. Codec - support decoder(mp3, aac).
5. Resampler - change sampling rate.
6. Add Media Debug.

Signed-off-by: jaesick.shin <jaesick.shin@samsung.com>

### CODE REVIEW

File: apps/examples/mediarecorder/mediarecorder_main.cpp [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+//***************************************************************************
+// examples/mediarecorder/mediarecorder_main.cxx
```
Comment (Taejun-Kwon) [inline apps/examples/mediarecorder/mediarecorder_main.cpp]: Is This corect copyright???

Comment (jsdosa) [reply apps/examples/mediarecorder/mediarecorder_main.cpp]: fixed it.

File: build/configs/artik055s/audio/Make.defs [hunk 2]
```diff
@@ -62,14 +62,14 @@ ifeq ($(WINTOOL),y)
   DIRUNLINK = $(TOPDIR)/tools/unlink.sh
   MKDEP = $(TOPDIR)/tools/mkwindeps.sh
   ARCHINCLUDES = -I. -isystem "${shell cygpath -w $(TOPDIR)/include}" -isystem "${shell cygpath -w $(TOPDIR)/../framework/include}" -isystem "${shell cygpath -w $(TOPDIR)/../external/include}"
-  ARCHXXINCLUDES = -I. -isystem "${shell cygpath -w $(TOPDIR)/include}" -isystem "${shell cygpath -w $(TOPDIR)/../external/include}"
-  ARCHSCRIPT = -T "${shell cygpath -w $(TOPDIR)/../build/configs/artik055s/scripts/$(LDSCRIPT)}"
+  ARCHXXINCLUDES = -I. -isystem "${shell cygpath -w $(TOPDIR)/include}" -isystem "${shell cygpath -w $(TOPDIR)/../framework/include}" -isystem "${shell cygpath -w $(TOPDIR)/../external/include}" -isystem "${shell cygpath -w $(TOPDIR)/../external/include/libcxx}"
+  ARCHSCRIPT = -T "${shell cygpath -w $(TOPDIR)/../build/configs/artik053/scripts/$(LDSCRIPT)}"
```
Comment (sunghan-chang) [inline build/configs/artik055s/audio/Make.defs]: ```artik055s``` is right

File: build/configs/artik055s/audio/Make.defs [hunk 3]
```diff
@@ -62,14 +62,14 @@ ifeq ($(WINTOOL),y)
...
 else
   # Linux/Cygwin-native toolchain
   MKDEP = $(TOPDIR)/tools/mkdeps$(HOSTEXEEXT)
   ARCHINCLUDES = -I. -isystem $(TOPDIR)/include -isystem $(TOPDIR)/../framework/include -isystem $(TOPDIR)/../external/include
-  ARCHXXINCLUDES = -I. -isystem $(TOPDIR)/include -isystem $(TOPDIR)/../external/include
-  ARCHSCRIPT = -T$(TOPDIR)/../build/configs/artik055s/scripts/$(LDSCRIPT)
+  ARCHXXINCLUDES = -I. -isystem $(TOPDIR)/include -isystem $(TOPDIR)/../framework/include -isystem $(TOPDIR)/../external/include -isystem $(TOPDIR)/../external/include/libcxx
+  ARCHSCRIPT = -T$(TOPDIR)/../build/configs/artik053/scripts/$(LDSCRIPT)
```
Comment (sunghan-chang) [inline build/configs/artik055s/audio/Make.defs]: same as previous

File: apps/examples/mediaplayer/Kconfig [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#
+# For a description of the syntax of this configuration file,
+# see kconfig-language at https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt
+#
+
+config EXAMPLES_MEDIAPLAYER
+	bool "MediaPlayer example"
+	default n
+	depends on HAVE_CXX && HAVE_CXXINITIALIZE
```
Comment (sunghan-chang) [inline apps/examples/mediaplayer/Kconfig]: Is this enough? Don't you need media or audio config?

**Comment by sunghan-chang** [comment]
@jsdosa You should check the location of calling codec's Kconfig. I can't find.

**Pull request merged**